File: src/ItemCollection.java
import java.awt.image.BufferedImage;
import java.io.Serializable;

public class ItemCollection implements Serializable {

    String type;
    short[] ids, nums, durs;
    transient BufferedImage image;
    double FUELP = 0;
    double SMELTP = 0;
    boolean F_ON = false;
    short recipeNum;

    public ItemCollection(String type, short[] ids, short[] nums, short[] durs) {
        this.type = type;
        this.ids = ids;
        this.nums = nums;
        this.durs = durs;
    }
}

File: src/main/scala/org/terraframe/ItemCollection.scala
package org.terraframe

import java.awt.image._
import java.io.Serializable

case class ItemCollection(`type`: String, ids: Array[Short], nums: Array[Short], durs: Array[Short]) extends Serializable {
    @transient var image: BufferedImage = _
    var FUELP: Double = 0
    var SMELTP: Double = 0
    var F_ON: Boolean = false
    var recipeNum: Short = _
}
